helper/proxyutil/proxyutil.go

```diff
@@ -0,0 +1,104 @@
+package proxyutil
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	proxyproto "github.com/armon/go-proxyproto"
+	"github.com/hashicorp/errwrap"
+	sockaddr "github.com/hashicorp/go-sockaddr"
+	"github.com/hashicorp/vault/helper/strutil"
+)
+
+// ProxyProtoConfig contains configuration for the PROXY protocol
+type ProxyProtoConfig struct {
+	sync.RWMutex
+	Behavior     string
+	AllowedAddrs []*sockaddr.SockAddrMarshaler `json:"allowed_addrs"`
+}
+
+func (p *ProxyProtoConfig) SetAllowedAddrs(addrs interface{}) error {
```

Comment:
This method could really use some unit tests to ensure the allowed addresses are being set correctly.